lib/ecto/adapters/myxql/connection.ex

```diff
@@ -683,6 +692,10 @@ if Code.ensure_loaded?(MyXQL) do
       ["NOT (", expr(expr, sources, query), ?)]
     end
 
+    defp expr({bnot, _, [expr]}, sources, query) when bnot in ~w(~~~ bnot)a do
+      ["~" | op_to_binary(expr, sources, query)]
```

Comment:
Not sure if `op_to_binary` is right here. The function seems to do the same job as `maybe_paren` does in the postgres connection, but its name suggests it should be used only for binary operators, which `bnot` is not

Reply:
i don't believe we can support bitstrig for mysql. it seems varbinary/binary refer to byte strings https://dev.mysql.com/doc/refman/8.0/en/binary-varbinary.html

Reply:
I'm not sure either. My plan was to add integration tests to observe how MySQL behaves with `varbinary` and then decide whether to keep or remove this feature for MySQL.

MySQL does have some support for bit operations though ( https://dev.mysql.com/doc/refman/8.0/en/bit-functions.html ). My guess is that we could add support for bitstring in MySQL, but limit it to bitstrings with segments size of 8 bits (a byte). 

Would that be acceptable?

Reply:
I believe we already handle that with `:binary`

```elixir
{:add, :token, :binary, [size: 20, null: false]},
```

Reply:
Sorry I got mixed up where I was commenting =/. I was referring to the migrations. I don't think we should add a `:bitstring` migration for MySQL. But yeah it seems to make sense to enable the operators.

Reply:
So to be clear, I think we need to change this

```elixir
defp ecto_to_db(:bitstring, _query), do: "varbinary"
```

to this

```elixir
defp ecto_to_db(:bitstring, _query), do: error!(query, "bitstring type is not supported by MySQL")
```

Reply:
This will cover the cases when someone is trying to create migrations with `:bitstring` or cast to `:bitstring`.

Reply:
There is actually a `bit` type in mysql lhttps://dev.mysql.com/doc/refman/8.0/en/bit-type.html

Reply:
Ok, this makes sense.

In my opinion, though, it would be preferable to keep the `:bitstrign` Ecto type for MySQL migrations. This ensures uniformity in the schema definition, where we have the `:bitstring` type, and maintains consistency across different database systems.

We could either use the `:bitstring` type as an alias of `:binary` or utilize the existing `bit` type in MySQL.

Reply:
I think using the `bit` type makes most sense.